### PR TITLE
Category[Is] and related tests

### LINKED ISSUE #2720: Category instance for Is
Is there any reason why **cats.evidence.Is** isn't a **Category**?
It's already have **compose** and **id** is **Is.refl**

### PR DESCRIPTION
**Pull request by ctongfei**
This should fix #2720 .

### CODE REVIEW

**Comment by kailuowang** [comment]
Thanks! There is just Scalafmt failure. Would you run sbt prePR and commit the changes ?

**Comment by ctongfei** [comment]
@kailuowang Thanks! I'll figure it out soon.

**Comment by ctongfei** [comment]
@kailuowang This should be OK now.

File: core/src/main/scala/cats/evidence/Is.scala [hunk 1]
```diff
@@ -65,7 +66,19 @@ abstract class Is[A, B] extends Serializable {
...
 }
 
-object Is {
+sealed abstract class IsInstances {
+  import Is._
+
+  /**
+   * The category instance on Leibniz categories.
+   */
+  implicit val leibniz: Category[Is] = new Category[Is] {
```
Comment (LukaJCB) [inline core/src/main/scala/cats/evidence/Is.scala]: Can we give thise one a more verbose name? Something like `catsEvidenceCategoryForLeibniz`? :)

Comment (ctongfei) [reply core/src/main/scala/cats/evidence/Is.scala]: The `Category` instance for `As` is named as `liskov` -- so I copied that :-)

Comment (LukaJCB) [reply core/src/main/scala/cats/evidence/Is.scala]: Good call :)

**Pull request merged**